### ISSUE TITLE
pointermove events on Windows crash with InvalidType

### ISSUE THREAD

**Issue by esimkowitz** [DioxusLabs/dioxus#2385]
**Problem**

The `onpointermove` event appears broken on Windows. When a component that binds to this event is present, the moment the user moves the cursor, the main thread panics with an InvalidType error. 

**Steps To Reproduce**

- Create a component that binds to the `onpointermove` event.
- On Windows, run `dx serve --platform desktop`
- Access the component with the bound event
- Try moving the mouse
- Observe that the application crashes

**Expected behavior**

Binding to the `onpointermove` event should return an `Event<PointerData>` event to the event handler.

**Screenshots**

N/A

**Environment:**
 - Dioxus version: 0.5.1
 - Rust version: 1.78.0
 - OS info: Windows 11 (22631.3527)
 - App platform: desktop

**Questionnaire**

- [ ] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by DogeDark** [comment]
I am unable to reproduce it on Windows 10. Perhaps it's a Windows 11 issue? I used this code snippet (is this what causes the crash for you?)

```rust
rsx! {
    div {
        style: "width: 10vw; height: 10vh; background-color: red;",
        onpointermove: move |data| {
            println!("hi | {:?}", data);
        }
    }
}
```

**Comment by esimkowitz** [comment]
> I am unable to reproduce it on Windows 10. Perhaps it's a Windows 11 issue? I used this code snippet (is this what causes the crash for you?)

Yeah this is what caused the issue for me. I can try to dig into it a bit more and see if there's some field that is getting added for Windows 11 that we're missing.

**Comment by Doublonmousse** [comment]
Same issue here, It segfaults on any input (mouse or pen input) on windows 11.

**Comment by Doublonmousse** [comment]
All fields are named correctly, the issue is types

**Comment by Doublonmousse** [comment]
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">




</head>

<body link="#467886" vlink="#96607D">


name | type expected by dioxus | windows 11, pen hover and mouse | windows 11, touch press | windows 11, pen press | guess
-- | -- | -- | -- | -- | --
  |   |   |   |   |  
alt_key | bool | bool | bool | bool | bool
button | i16 | i64 | i64 | i64 | i64
buttons | u16 | u64 | u64 | u64 | u64
client_x | i32 | f64 | f64 | u64 | f64
client_y | i32 | u64 | f64 | f64 | f64
ctrl_key | bool | bool | bool | bool | bool
height | i32 | u64 | u64 | u64 | u64
is_primary | bool | bool | bool | bool | bool
meta_key | bool | bool | bool | bool | bool
offset_x | i32 | f64 | f64 | f64 | f64
offset_y | i32 | u64 | f64 | f64 | f64
page_x | i32 | f64 | f64 | u64 | f64
page_y | i32 | u64 | f64 | f64 | f64
pointer_id | i32 | u64 | u64 | u64 | u64
pointer_type | string | string | string | string | string
pressure | f32 | u64 | f64 | f64 | f64
screen_x | i32 | f64 | f64 | u64 | f64
screen_y | i32 | u64 | f64 | f64 | f64
shift_key | bool | bool | bool | bool | bool
tangential_pressure | f32 | u64 | u64 | u64 | f64
tilt_x | i32 | u64 | u64 | u64 | i64
tilt_y | i32 | u64 | u64 | u64 | i64
twist | i32 | u64 | u64 | u64 | u64
width | i32 | u64 | f64 | u64 | f64



</body>

</html>

On the left is the types defined in dioxus, the other columns are the reported types when the error message occurs and on the right is the guess for the correct types values on windows 11.

**Comment by Doublonmousse** [comment]
Ok, I have a working version here https://github.com/Doublonmousse/dioxus for windows 11.

Type changes are of the types `i16 -> i64`, `i32 -> i64`, `i32 -> u64` (there may be ways to cast the values when deserializing)
but there's a couple of `i32 -> f64` changes that'd be a little more challenging.

And some `f32 -> f64` too

I'm not sure the debug print for `onpointermove` shows everything I'd want to test yet (are all button presses and release events captured for a pen that's hovering? Same thing for a pen on the screen. Is the eraser correctly detected ?)

**Comment by jkelleyrtp** [comment]
We now handle events without pancking - just fail to deserialize - but we really need to fix the underlying bug.

It doesn't look like we'll be able to fix it for 0.6.0 release but might be able to later during the 0.6.0 release cycle.

**Comment by vascokk** [comment]
Same issue on Android ("expected i32"). Dioxus 0.6.3:
```
RustStdoutStderr: ERROR Error parsing user_event: Error("Failed to deserialize event data for event pointerup:  Error(\"invalid type: floating point `187.96875`, expected i32\", line: 0, column: 0)\n'Object {\n    \"alt_key\": Bool(false),\n    \"button\": Number(0),\n    \"buttons\": Number(0),\n    \"client_x\": Number(187.96875),\n    \"client_y\": Number(602.9775390625),\n    \"ctrl_key\": Bool(false),\n    \"height\": Number(1),\n    \"is_primary\": Bool(true),\n    \"meta_key\": Bool(false),\n    \"offset_x\": Number(166.96875),\n    \"offset_y\": Number(105.8525390625),\n    \"page_x\": Number(187.96875),\n    \"page_y\": Number(1029.9775390625),\n    \"pointer_id\": Number(45),\n    \"pointer_type\": String(\"touch\"),\n    \"pressure\": Number(0),\n    \"screen_x\": Number(187.96875),\n    \"screen_y\": Number(626.9775390625),\n    \"shift_key\": Bool(false),\n    \"tangential_pressure\": Number(0),\n    \"tilt_x\": Number(0),\n    \"tilt_y\": Number(0),\n    \"twist\": Number(0),\n    \"width\": Number(1),\n}'", line: 0, column: 0).Contents: Ok(""), raw: Request {
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:     method: POST,
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:     uri: dioxus://index.html/__events,
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:     version: HTTP/1.1,
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:     headers: {
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:         "user-agent": "Mozilla/5.0 (Linux; Android 16; sdk_gphone64_x86_64 Build/BP22.250325.006; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/133.0.6943.137 Safari/537.36",
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:         "origin": "http://dioxus.index.html",
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:         "referer": "http://dioxus.index.html/",
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:         "accept": "*/*",
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:         "dioxus-data": "eyJuYW1lIjoicG9pbnRlcnVwIiwiZGF0YSI6eyJhbHRfa2V5IjpmYWxzZSwiYnV0dG9uIjowLCJidXR0b25zIjowLCJjbGllbnRfeCI6MTg3Ljk2ODc1LCJjbGllbnRfeSI6NjAyLjk3NzUzOTA2MjUsImN0cmxfa2V5IjpmYWxzZSwibWV0YV9rZXkiOmZhbHNlLCJvZmZzZXRfeCI6MTY2Ljk2ODc1LCJvZmZzZXRfeSI6MTA1Ljg1MjUzOTA2MjUsInBhZ2VfeCI6MTg3Ljk2ODc1LCJwYWdlX3kiOjEwMjkuOTc3NTM5MDYyNSwic2NyZWVuX3giOjE4Ny45Njg3NSwic2NyZWVuX3kiOjYyNi45Nzc1MzkwNjI1LCJzaGlmdF9rZXkiOmZhbHNlLCJwb2ludGVyX2lkIjo0NSwid2lkdGgiOjEsImhlaWdodCI6MSwicHJlc3N1cmUiOjAsInRhbmdlbnRpYWxfcHJlc3N1cmUiOjAsInRpbHRfeCI6MCwidGlsdF95IjowLCJ0d2lzdCI6MCwicG9pbnRlcl90eXBlIjoidG91Y2giLCJpc19wcmltYXJ5Ijp0cnVlfSwiZWxlbWVudCI6MTYsImJ1YmJsZXMiOnRydWV9",
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:         "content-type": "application/json",
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:     },
04-27 20:15:30.331  6131  6178 I RustStdoutStderr:     body: [],
04-27 20:15:30.331  6131  6178 I RustStdoutStderr: }

```
